src/app/profile/yourGiving/stopStartRecurringGifts/restartGift/restartGift.component.js

```diff
@@ -51,6 +54,11 @@ class RestartGiftController {
         this.paymentMethods = paymentMethods;
         this.nextDrawDate = nextDrawDate;
         this.hasPaymentMethods = paymentMethods && paymentMethods.length > 0;
+        if(!this.hasPaymentMethods) {
```

Comment:
```javascript
         RecurringGiftModel.paymentMethods = this.validPaymentMethods;
         RecurringGiftModel.nextDrawDate = this.nextDrawDate;
```
This adds the list of valid (non expired) payment methods and nextDrawDate to the RecurringGiftModel. This is a class property that is used by each instance of RecurringGiftModel to either display payment information or allow editing payment information. If you add a new valid payment method, it should get added to this array or it won't show up in the list of available methods during the configureGift section.